src/webui/abstractwebapplication.h

```diff
@@ -75,24 +68,21 @@ class AbstractWebApplication : public Http::ResponseBuilder, public Http::IReque
     static QString saveTmpFile(const QByteArray &data);
 
     WebSessionData *session();
-    const Http::Request &request() const { return request_; }
```

Comment:
instead of moving to `WebApplication`, why not move to protected section of `AbstractWebApplication`?

Reply:
@Chocobo1, this PR has currently some unfinished/intermediate changes. Please don't review until I finish it.